src/Injector.hack

```diff
@@ -112,31 +113,18 @@ abstract class HTMLPurifier_Injector {
 	 */
 	public function checkNeeded(HTMLPurifier_Config $config): string {
 		$def = TypeAssert\matches<Definition\HTMLPurifier_HTMLDefinition>($config->getHTMLDefinition());
-		$i = 0;
-		$index = 0;
 		foreach ($this->needed as $element => $attributes) {
-			if ($i is int) {
-				$i = $element;
+			if (!HH\idx($def->info, $element)) {
+				return $element;
 			}
-			if ($i is string && !C\contains_key($def->info, $i)) {
-				return $i;
-			}
-			if (!($element is vec<_>)) {
-				$index++;
-				$i = $index;
+			if (!($attributes is vec<_>)) {
```

Comment:
Does the typechecker require this? Seems like this should be guaranteed to always be a `vec` thanks to the type of `needed`?

Reply:
Nope - ✂️!